test/integration/scheduler/util.go

```diff
@@ -156,6 +159,28 @@ func createNodes(cs clientset.Interface, prefix string, wrapper *st.NodeWrapper,
 	return nodes[:], nil
 }
 
+// createAndWaitForNodesInCache calls createNodes(), and wait for the created
+// nodes to be present in scheduler cache.
+func createAndWaitForNodesInCache(testCtx *testutils.TestContext, prefix string, wrapper *st.NodeWrapper, numNodes int) ([]*v1.Node, error) {
+	nodes, err := createNodes(testCtx.ClientSet, prefix, wrapper, numNodes)
+	if err != nil {
+		return nodes, fmt.Errorf("fail to create nodes: %v", err)
+	}
+	return nodes, waitForNodesInCache(testCtx.Scheduler, numNodes)
```

Comment:
This function won't work when it's called more than once.

What you can do is obtain the node counter before calling createNodes, and pass that to `waitForNodesInCache`, plus `numNodes`.

Reply:
Please see if the new commit does what you expect.